File: program/program/cylindrically-rotate-a-matrix-by-90-degrees/cylindrically_rotate_a_matrix_by_90_degrees.fs
let cylindricalRotate90 (matrix:int[,]) =
    let rows = matrix.GetLength(0)
    let cols = matrix.GetLength(1)
    
    // Create a new matrix to store the rotated values
    let result = Array2D.create cols rows 0
    
    // Rotate the matrix by 90 degrees
    for i in 0..(rows - 1) do
        for j in 0..(cols - 1) do
            result.[j, rows - 1 - i] <- matrix.[i, j]
    
    result

// Define the input matrix
let inputMatrix = array2D [[1; 2; 3]; [4; 5; 6]; [7; 8; 9]]

// Print the original matrix
printfn "Input Matrix:"
for i in 0..(inputMatrix.GetLength(0) - 1) do
    for j in 0..(inputMatrix.GetLength(1) - 1) do
        printf "%d " inputMatrix.[i, j]
    printfn ""

// Perform cylindrical rotation
let rotatedMatrix = cylindricalRotate90 inputMatrix

// Print the rotated matrix
printfn "\nOutput Matrix:"
for i in 0..(rotatedMatrix.GetLength(0) - 1) do
    for j in 0..(rotatedMatrix.GetLength(1) - 1) do
        printf "%d " rotatedMatrix.[i, j]
    printfn ""

File: program/program/cylindrically-rotate-a-matrix-by-90-degrees/cylindrically_rotate_a_matrix_by_90_degrees.R
# Function to perform cylindrical rotation
cylindrical_rotate <- function(matrix_input) {
  # Get the number of rows and columns in the matrix
  num_rows <- nrow(matrix_input)
  num_cols <- ncol(matrix_input)

  # Create an empty matrix to store the rotated result
  rotated_matrix <- matrix(0, nrow = num_cols, ncol = num_rows)

  # Perform the cylindrical rotation
  for (i in 1:num_rows) {
    rotated_matrix[, i] <- matrix_input[num_rows - i + 1, ]
  }

  return(rotated_matrix)
}

# Example usage
input_matrix <- matrix(c(1, 2, 3, 4, 5, 6, 7, 8, 9), nrow = 3, byrow = TRUE)

cat("Input Matrix:\n")
print(input_matrix)

rotated_matrix <- cylindrical_rotate(input_matrix)

cat("\nRotated Matrix:\n")
print(rotated_matrix)
